docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
 ---
 
 # Streams
-You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
+Streams are the most common type of live collection in Hydro; they can be used to model streaming data collections, a feed of API requests, or even time-based intervals. A `Stream` represents a sequence of elements, with new elements being asynchronously appended to the end of the sequence. Streams can be transformed using APIs like `map` and `filter`, based on Rust [iterators](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html). You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
+
+Streams have several type parameters:
+- `T`: the type of elements in the stream
+- `L`: the location the stream is on (see [Locations](../locations/index.md))
+- `B`: indicates whether the stream is [bounded or unbounded](./bounded-unbounded.md)
+- `Order`: indicates whether the elements in the stream have a deterministic order or not
+  - This type parameter is _optional_; by default the order is deterministic
+
+## Creating a Stream
+The simplest way to create a stream is to use the [`source_iter`](https://hydro.run/rustdoc/hydro_lang/location/trait.Location#method.source_iter) method on a location, which creates a stream from any Rust type that can be converted into an [`Iterator`](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html) (via [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html)). For example, we can create a stream of integers on a [process](../locations/processes.md) and transform it:
```

Comment:
what does it mean to "use a method *on a location*"? That's a bit opaque.

Reply:
Maybe instead say "`source_iter` converts any Rust iterator over type `T` into a `Stream<T, L, Bounded, TotalOrder>`, where the Location `L` is ...." and forward ref the `TotalOrder` part to the section below.

Reply:
Are we able to assert that all Rust iterators generate `Bounded` streams? what about `repeat()` and the like?